examples/src/bin/ech-client.rs

```diff
@@ -65,11 +75,15 @@ async fn main() {
         .parse_filters("trace")
         .init();
 
-    let ech_mode = match server_ech_config {
-        Some(ech_config_list) => EchConfig::new(ech_config_list, ALL_SUPPORTED_SUITES)
-            .unwrap()
-            .into(),
-        None => {
+    // If the ECH config list is empty, use a GREASE ECH config.
+    // Otherwise, use the first compatible ECH config from one of the ECH config lists.
+    let ech_mode = match server_ech_configs.is_empty() {
+        false => server_ech_configs
+            .into_iter()
+            .find_map(|list| EchConfig::new(list, ALL_SUPPORTED_SUITES).ok())
+            .map(Into::into)
```

Comment:
I feel like the use of `Into` is a little opaque here (also this particular case could/should probably go inside the `find_map()`?) -- the type we're casting to isn't really clear here, so maybe go with some kind of explicit conversion (even just wrapping in `T::from()`)?

Reply:
Fixed up.